appservice/src/editScmType.ts

```diff
@@ -23,10 +23,18 @@ export async function editScmType(client: SiteClient, node: AzureTreeItem): Prom
         await connectToGitHub(node, client);
     } else {
         config.scmType = newScmType;
-        // to update one property, a complete config file must be sent
+        // to update one property, a complete config file must be sentnpm
         await client.updateConfiguration(config);
     }
     ext.outputChannel.appendLine(localize('deploymentSourceUpdated,', 'Deployment source for "{0}" has been updated to "{1}".', client.fullName, newScmType));
+    if (newScmType === ScmType.LocalGit) {
+        const user: WebSiteManagementModels.User = await client.getPublishingUser();
+        if (user.publishingUserName) {
+            // first time users must set up deployment credentials via the Portal or they will not have a UserName
+            const gitCloneUri: string = `https://${user.publishingUserName}@${client.gitUrl}`;
```

Comment:
What's the relationship between this uri and publishCredentials.scmUri?

Reply:
The username from `getPublishingUser` is universally accepted on a subscription level.  The `publishCredentials` username and scmUri are both on the app level.  If users use that uri and enter the credentials they set on the portal for user level access, it'll fail authentication so we have to provide them with the proper username.